File: backend/routes/sessions.js
const express = require('express');
const router = express.Router();
const { Session, TranscriptChunk, Summary, ActionItem, Concept } = require('../models');
const { processVideoAsync } = require('../services/processor');
const manager = require('../services/sockets');

// POST /api/v1/sessions/process
router.post('/process', async (req, res) => {
  try {
    const { url, title } = req.body;
    
    // Create session record
    const session = await Session.create({
      title: title || "New Analysis",
      source_url: url,
      source_type: "youtube",
      status: "pending"
    });

    // Spawn background task (don't await)
    processVideoAsync(session.id, manager);

    res.status(201).json(session);
  } catch (error) {
    console.error("Error creating session:", error);
    res.status(500).json({ error: error.message });
  }
});

// GET /api/v1/sessions/:id
router.get('/:id', async (req, res) => {
  try {
    const session = await Session.findByPk(req.params.id, {
      include: [
        { model: TranscriptChunk, as: 'transcript_chunks' },
        { model: Summary, as: 'summaries' },
        { model: ActionItem, as: 'action_items' },
        { model: Concept, as: 'concepts' }
      ]
    });

    if (!session) {
      return res.status(404).json({ error: "Session not found" });
    }

    // Format summaries for frontend convenience
    const summaryMap = {};
    (session.summaries || []).forEach(s => {
      summaryMap[s.mode] = s.content;
    });

    res.json({
      id: session.id,
      title: session.title,
      status: session.status,
      source_url: session.source_url,
      transcript: session.transcript_chunks,
      summaries: summaryMap,
      action_items: session.action_items,
      concepts: session.concepts
    });
  } catch (error) {
    console.error("Error fetching session:", error);
    res.status(500).json({ error: error.message });
  }
});

module.exports = router;

File: backend/app/api/v1/endpoints/sessions.py
from fastapi import APIRouter, Depends, HTTPException, BackgroundTasks
from sqlalchemy.orm import Session
from typing import Any, List
from app.db.session import SessionLocal
from app.models.session import Session as SessionModel
from app.worker.tasks import process_video_async

router = APIRouter()

def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

@router.post("/process")
async def process_video(
    *,
    db: Session = Depends(get_db),
    url: str,
    title: str = "New Analysis",
    background_tasks: BackgroundTasks
) -> Any:
    """
    Creates a session and triggers background processing.
    """
    # 1. Create session record
    session = SessionModel(
        title=title,
        source_url=url,
        source_type="youtube",
        status="pending"
    )
    db.add(session)
    db.commit()
    db.refresh(session)
    
    # 2. Trigger background task
    background_tasks.add_task(process_video_async, session.id)
    
    return session

@router.get("/{session_id}")
async def get_session(
    session_id: int,
    db: Session = Depends(get_db)
) -> Any:
    """
    Gets full session data.
    """
    session = db.query(SessionModel).filter(SessionModel.id == session_id).first()
    if not session:
        raise HTTPException(status_code=404, detail="Session not found")
        
    return {
        "id": session.id,
        "title": session.title,
        "status": session.status,
        "source_url": session.source_url,
        "transcript": session.transcript_chunks,
        "summaries": {s.mode: s.content for s in session.summaries},
        "action_items": session.action_items,
        "concepts": session.concepts
    }
